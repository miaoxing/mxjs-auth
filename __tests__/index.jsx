import 'core-js/stable/structured-clone';
import {AuthProvider, useAuth} from '../index';
import {render} from '@testing-library/react';

describe('auth', () => {
  test('basic', () => {
    const Child = () => {
      const {access} = useAuth();
      return access('page/sub-page') ? '1' : '0';
    };

    const {container} = render(<AuthProvider permissions={{
      'page/sub-page': true,
    }}>
      <Child/>
    </AuthProvider>);
    expect(container).toMatchSnapshot();
  });
});
