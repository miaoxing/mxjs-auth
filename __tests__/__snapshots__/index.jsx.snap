// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth basic 1`] = `
<div>
  1
</div>
`;
